docs/source/links/state.md

```diff
@@ -0,0 +1,295 @@
+   you can set them from [within your
+   component](docs/react/basics/queries.html#graphql-config-options-context) or
+   by [using `apollo-link-context`](/docs/link/links/context.html).
+4. `info`: Information about the execution state of the query. You will probably
+   never have to use this one.
+
+For further exploration, check out the [`graphql-tools`
+docs](/docs/graphql-tools/resolvers.html#Resolver-function-signature).
+
+<h3 id="async">Async resolvers</h3>
+
+`apollo-link-state` supports asynchronous resolver functions. This can be useful
+for performing side effects like accessing a device API. If you would like to
+hit a REST endpoint with your resolver, [we recommend checking out
+`apollo-link-rest`](https://github.com/apollographql/apollo-link-rest) instead,
+which is a more complete solution for using your REST endpoints with Apollo
+Client.
+
+For React Native and most browser APIs, you should set up a listener in a
+component lifecycle method and pass in your mutation trigger function as the
+callback instead of using an async resolver. However, there are some cases where
+it's beneficial to perform the side effect within a resolver:
+
+```js
+import { CameraRoll } from 'react-native';
+
+const cameraRoll = {
+  Query: {
+    cameraRoll: async (_, { assetType }) => {
+      try {
+        const media = await CameraRoll.getPhotos({
+          first: 20,
+          assetType,
+        });
+
+        return {
+          ...media,
+          id: assetType,
+          __typename: 'CameraRoll',
+        };
+      } catch (e) {
+        console.error(e);
+        return null;
+      }
+    },
+  },
+};
+```
+
+[`CameraRoll.getPhotos()`](https://facebook.github.io/react-native/docs/cameraroll.html#getphotos)
+returns a Promise resolving to an object with a `edges` property, which is an
+array of camera node objects, and a `page_info` property, which is an object
+with pagination information. This is a great use case for GraphQL, since we can
+filter down the return value to only the data that our components consume.
+
+```js
+const GET_PHOTOS = gql`
+  query getPhotos($assetType: String!) {
+    cameraRoll(assetType: $assetType) {
+      id
+      edges {
+        node {
+          image {
+            uri
+          }
+          location {
+            latitude
+            longitude
+          }
+        }
+      }
+    }
+  }
+`;
+```
+
+<h3 id="organize">Organizing resolver maps</h3>
+
+For most applications, your resolver map will probably be too large to fit in
+one file. To organize your resolver map, we recommend splitting it up by
+feature, similar to the Redux [ducks
+pattern](https://github.com/erikras/ducks-modular-redux). Each feature will have
+its own set of queries, mutations, and fields on its resolver map. Then, you can
+merge all of your separate resolver maps into one object before you pass it to
+`withClientState`.
+
+```js
+import { merge } from 'lodash';
```

Comment:
I'm worried about suggesting people use lodash on the client; is there a simpler merge implementation we can recommend?

Reply:
maybe `lodash.merge`?

Reply:
That is still mega huge. I'm kinda allergic to anything with the word `lodash` in it after fighting it earlier this year. I could write a 10-line replacement for merge and include it in link-state if that helps